spec/controllers/lti_deployment_controller_spec.rb

```diff
@@ -1,136 +1,7 @@
-          end
-        end
-      end
-    end
-  end
-  describe '#redirect_login' do
-    before :each do
-      controller.request.headers.merge(HTTP_REFERER: host)
-    end
-    context 'with incorrect or missing parameters' do
-      it 'redirects to an error page with no params' do
-        post :redirect_login, params: {}
-        is_expected.to render_template('shared/http_status')
-      end
-      it 'redirects to an error page with a mismatched state' do
-        post :redirect_login, params: { state: state, id_token: 'token' }
-        is_expected.to render_template('shared/http_status')
-      end
-    end
-    context 'with correct parameters' do
-      let(:jwk_url) { 'https://canvas.instructure.com:443/api/lti/security/jwks' }
-      let(:payload) do
-        { aud: client_id,
-          iss: 'https://canvas.instructure.com',
-          'https://purl.imsglobal.org/spec/lti/claim/deployment_id': 'some_deployment_id',
-          'https://purl.imsglobal.org/spec/lti/claim/context': {
-            label: 'csc108',
-            title: 'test'
-          },
-          'https://purl.imsglobal.org/spec/lti/claim/custom': {
-            course_id: 1,
-            user_id: 1
-          } }
-      end
-      let(:pub_jwk) { JWT::JWK.new(OpenSSL::PKey::RSA.new(1024)) }
-      let(:lti_jwt) { JWT.encode(payload, pub_jwk.keypair, 'RS256', { kid: pub_jwk.kid }) }
-      before :each do
-        session[:client_id] = client_id
-        stub_request(:get, jwk_url).to_return(status: 200, body: { keys: [pub_jwk.export] }.to_json)
-      end
-      it 'successfully decodes the jwt and redirects' do
-        post :redirect_login, params: { state: session.id.to_s, id_token: lti_jwt }
-        expect(response).to redirect_to(root_path)
-      end
-      it 'successfully decodes the jwt and sets lti_course_id in the session' do
-        post :redirect_login, params: { state: session.id.to_s, id_token: lti_jwt }
-        expect(session[:lti_course_id]).to eq(1)
-      end
-      it 'successfully decodes the jwt and sets lti_course_name in the session' do
-        post :redirect_login, params: { state: session.id.to_s, id_token: lti_jwt }
-        expect(session[:lti_course_name]).to eq('test')
-      end
-      it 'successfully decodes the jwt and sets lti_course_label in the session' do
-        post :redirect_login, params: { state: session.id.to_s, id_token: lti_jwt }
-        expect(session[:lti_course_label]).to eq('csc108')
-      end
-      it 'successfully decodes the jwt and sets lti_user_id in the session' do
-        post :redirect_login, params: { state: session.id.to_s, id_token: lti_jwt }
-        expect(session[:lti_user_id]).to eq(1)
-      end
-      it 'successfully creates a new lti object' do
-        post :redirect_login, params: { state: session.id.to_s, id_token: lti_jwt }
-        expect(LtiDeployment.count).to eq(1)
-      end
-    end
-  end
+
   describe '#choose_course', :choose_course do
```

Comment:
It can be used for [metadata](https://relishapp.com/rspec/rspec-core/docs/metadata/user-defined-metadata) - but in this case it was an artifact of trying to get a test to work before realizing I had misconfigured routes and doesn't do anything, so I've removed it.